src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,207 @@
+        Set<LocalizedAlert> resolvedAlerts = new HashSet<>(previousAlerts);
+        resolvedAlerts.removeAll(newAlerts);
+        // If journey state is already tracking alerts from previous checks, see if they have changed.
+        if (unseenAlerts.size() > 0 || resolvedAlerts.size() > 0) {
+            enqueueNotification(TripMonitorNotification.createAlertNotification(unseenAlerts, resolvedAlerts));
+        }
+    }
+
+    private void checkTripForDelays(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+    private void checkTripForItineraryChanges(MonitoredTrip trip, Itinerary itinerary) {
+        // TODO
+    }
+
+    private void sendNotification() {
+        if (notifications.size() == 0) {
+            // FIXME: Change log level
+            LOG.info("No notifications queued up. Skipping notify.");
+            return;
+        }
+        OtpUser otpUser = Persistence.otpUsers.getById(trip.userId);
+        if (otpUser == null) {
+            LOG.error("Cannot find user for id {}", trip.userId);
+            // TODO: Bugsnag / delete monitored trip?
+            return;
+        }
+        String name = trip.tripName != null ? trip.tripName : "Trip for " + otpUser.email;
+        String subject = name + " Notification";
+        StringBuilder body = new StringBuilder();
+        for (TripMonitorNotification notification : notifications) {
+            body.append(notification.body);
+        }
+        // FIXME: Change log level
+        LOG.info("Sending notification '{}' to user {}", subject, trip.userId);
+        // FIXME: This needs to be an enum.
+        switch (otpUser.notificationChannel.toLowerCase()) {
+            // TODO: Use medium-specific messages (i.e., SMS body should be shorter/phone friendly)
+            case "sms":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                break;
+            case "email":
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            case "all":
+                NotificationUtils.sendSMS(otpUser.phoneNumber, body.toString());
+                NotificationUtils.sendEmail(otpUser.email, subject, body.toString(), null);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void enqueueNotification(TripMonitorNotification tripMonitorNotification) {
+        notifications.add(tripMonitorNotification);
+    }
+
+    private boolean shouldSkipMonitoredTripCheck(MonitoredTrip trip) {
+        // Default to system's zone if one can't be inferred from trip destination.
+        ZoneId zoneId = ZoneId.systemDefault();
+        // TODO: Verify that this check doesn't cost anything.
+        Optional<ZoneId> zoneIdForCoordinates = getZoneIdForCoordinates(trip.to.lat, trip.to.lon);
+        if (zoneIdForCoordinates.isEmpty()) {
+            LOG.error("Could not find timezone for monitored trip: {}", trip.id);
```

Comment:
If the timezone can't be obtained from GPS coordinates, something really funky is going on especially since the `getZoneIdForCoordinates` should have worldwide coverage. But if it happens, I think it doesn't make sense to use the default system timezone. Instead, an error should be thrown.

Reply:
throw RuntimeException